modules/disk-uploader/pkg/vmexport/vmexport_test.go

```diff
@@ -0,0 +1,263 @@
+package vmexport_test
+
+import (
+	"os"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	fakek8sclient "k8s.io/client-go/kubernetes/fake"
+	"k8s.io/client-go/testing"
+
+	"github.com/golang/mock/gomock"
+	"github.com/kubevirt/kubevirt-tekton-tasks/modules/disk-uploader/pkg/vmexport"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	v1beta1 "kubevirt.io/api/export/v1beta1"
+	kubevirtfake "kubevirt.io/client-go/generated/kubevirt/clientset/versioned/fake"
+	"kubevirt.io/client-go/kubecli"
+)
+
+var _ = Describe("VMExport", func() {
+	var (
+		ctrl           *gomock.Controller
+		kubeClient     *fakek8sclient.Clientset
+		vmExportClient *kubevirtfake.Clientset
+		virtClient     kubecli.KubevirtClient
+	)
+
+	const (
+		namespace = "test-namespace"
+		name      = "test-vmexport"
+	)
+
+	BeforeEach(func() {
+		ctrl = gomock.NewController(GinkgoT())
+		kubeClient = fakek8sclient.NewSimpleClientset()
+		vmExportClient = kubevirtfake.NewSimpleClientset()
+
+		kubecli.GetKubevirtClientFromClientConfig = kubecli.GetMockKubevirtClientFromClientConfig
+		kubecli.MockKubevirtClientInstance = kubecli.NewMockKubevirtClient(ctrl)
+		kubecli.MockKubevirtClientInstance.EXPECT().CoreV1().Return(kubeClient.CoreV1()).AnyTimes()
+		kubecli.MockKubevirtClientInstance.EXPECT().VirtualMachineExport(namespace).Return(vmExportClient.ExportV1beta1().VirtualMachineExports(namespace)).AnyTimes()
+
+		virtClient, _ = kubecli.GetKubevirtClientFromClientConfig(nil)
+	})
+
+	Describe("CreateVirtualMachineExport", func() {
+		Context("when the VirtualMachineExport created successfully", func() {
+			BeforeEach(func() {
+				kubeClient.Fake.PrependReactor("get", "pods", func(action testing.Action) (bool, runtime.Object, error) {
+					_, ok := action.(testing.GetAction)
+					Expect(ok).To(BeTrue())
+
+					pod := &corev1.Pod{
+						ObjectMeta: metav1.ObjectMeta{
+							Name:      name,
+							Namespace: namespace,
+						},
+					}
+					return true, pod, nil
+				})
+				vmExportClient.Fake.PrependReactor("create", "virtualmachineexports", func(action testing.Action) (bool, runtime.Object, error) {
+					create, ok := action.(testing.CreateAction)
+					Expect(ok).To(BeTrue())
+
+					vmExport, ok := create.GetObject().(*v1beta1.VirtualMachineExport)
+					Expect(ok).To(BeTrue())
+					return true, vmExport, nil
+				})
+
+				os.Setenv("POD_NAME", name)
+				os.Setenv("POD_NAMESPACE", namespace)
+			})
+
+			AfterEach(func() {
+				os.Unsetenv("POD_NAME")
+				os.Unsetenv("POD_NAMESPACE")
+			})
+
+			DescribeTable("should return no error when created",
```

Comment:
nit - Very simple table. Can you consider using three `It()`s instead?

Reply:
I'd prefer to keep it as a table, because then with three `It()`s it's just more duplicated lines.